pkg/dependencies/factory.go

```diff
@@ -601,7 +615,7 @@ func (f *Factory) WithHelmSecure() *Factory {
 }
 
 func (f *Factory) WithHelmInsecure() *Factory {
-	f.WithExecutableBuilder()
+	f.WithExecutableBuilder().WithProxyConfigurationFromEnv()
```

Comment:
Also, if we call this here, does it mean that the values from env can override the values specified in cluster spec, or vice versa? If it's the other way, maybe it is okay? But ideally we can read the values from the clusterspec if we anyways need to take it in as input, but this would make it more consistent with how the docker pull command would read the env vars